### ISSUE TITLE
Build error on Ubuntu/Mint Linux: src/nd-sink.h no such file or directory

### ISSUE THREAD

**Issue by miurahr** [benzea/gnome-screencast#47]
When building on Ubuntu(bionic)/LinuxMint 19.1, it fails with following error:

```
[16/34] cc  -Isrc/gnome-network-displays@exe -Isrc -I../src -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libnm -I/usr/include/gstreamer-1.0 -I/usr/include/orc-0.4 -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread -D_REENTRANT -MD -MQ 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o' -MF 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o.d' -o 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o' -c src/nd-enum-types.c
FAILED: src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o 
cc  -Isrc/gnome-network-displays@exe -Isrc -I../src -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/libnm -I/usr/include/gstreamer-1.0 -I/usr/include/orc-0.4 -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -pthread -D_REENTRANT -MD -MQ 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o' -MF 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o.d' -o 'src/gnome-network-displays@exe/meson-generated_.._nd-enum-types.c.o' -c src/nd-enum-types.c
src/nd-enum-types.c:5:10: fatal error: src/nd-sink.h: No such file or directory
 #include "src/nd-sink.h"
          ^~~~~~~~~~~~~~~
compilation terminated.
```

It is happened on Launchpad PPA build platform, where is 
log: https://launchpad.net/~miurahr/+archive/ubuntu/wifi-p2p/+build/17332217

build configuration: https://launchpad.net/~miurahr/+archive/ubuntu/wifi-p2p/+packages

**Comment by miurahr** [comment]
Meson configuration is as follows;

```
	cd obj-x86_64-linux-gnu && LC_ALL=C.UTF-8 meson .. --wrap-mode=nodownload --buildtype=plain --prefix=/usr --sysconfdir=/etc --localstatedir=/var --libdir=lib/x86_64-linux-gnu --libexecdir=lib/x86_64-linux-gnu
The Meson build system
Version: 0.45.1
Source dir: /<<PKGBUILDDIR>>
Build dir: /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu
Build type: native build
Project name: gnome-network-displays
Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
Appending CFLAGS from environment: '-g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security'
Appending LDFLAGS from environment: '-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed'
Appending CPPFLAGS from environment: '-Wdate-time -D_FORTIFY_SOURCE=2'
Build machine cpu family: x86_64
Build machine cpu: x86_64
Configuring gnome-network-displays-config.h using configuration
Program desktop-file-validate found: NO
Program appstream-util found: NO
Program glib-compile-schemas found: YES (/usr/bin/glib-compile-schemas)
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency gstreamer-video-1.0 found: YES 1.14.4
Native dependency gstreamer-rtsp-1.0 found: YES 1.14.4
Native dependency gstreamer-rtsp-server-1.0 found: YES 1.14.4
Program glib-mkenums found: YES (/usr/bin/glib-mkenums)
Native dependency gio-2.0 found: YES 2.56.4
Native dependency gtk+-3.0 found: YES 3.22.30
Native dependency libnm found: YES 1.16.0
Native dependency gstreamer-1.0 found: YES 1.14.4
Native dependency gstreamer-pbutils-1.0 found: YES 1.14.4
Native dependency libpulse-mainloop-glib found: YES 11.1
Native dependency glib-2.0 found: YES 2.56.4
Program build-aux/meson/postinstall.py found: YES (/<<PKGBUILDDIR>>/build-aux/meson/postinstall.py)
Build targets in project: 11
Found ninja-1.8.2 at /usr/bin/ninja
```

**Comment by benzea** [comment]
Odd, I mean the file is just in the source directory, so it isn't picking it up correctly. I don't have a good idea right now why that might be happening though. Maybe some meson change that means that the include path is different?

**Comment by miurahr** [comment]
Here is paths  meson generated for compiling  `<build_dir>/src/nd-enum-types.c` on work directory `<build_dir>`
```
 -Isrc/gnome-network-displays@exe
 -Isrc
 -I../src
 -I/usr/include/glib-2.0  -I/usr/lib/x86_64-linux-gnu/glib-2.0/include <snip for dependency lib paths>  
 -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu 
``` 

now `<build_dir>` is `<root>/obj-x86_64-linux-gnu`

If line `#include "src/nd-sink.h"` is `#include "nd-sink.h"`, compile is succeeded.

**Comment by miurahr** [comment]
It is happened with meson 0.45.1 which is a version in Ubuntu LTS (Bionic), and also fails with 0.46.0.
When using meson 0.46.1, it works. So it is a bug of meson build system version fixed in meson 0.46.1.

**Comment by benzea** [comment]
I guess I should just bump the meson dependency to 0.46 then in the next tarball?
